Backend/auction-app/src/main/java/com/example/auctionapp/model/User.java

```diff
@@ -0,0 +1,90 @@
+package com.example.auctionapp.model;
+
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotBlank;
+import java.time.LocalDate;
+
+@Entity
+@Table(name = "user_entity")
+public class User extends Resource {
+
+    @NotBlank(message = "First name can't be blank")
+    @Column(name = "first_name")
+    private String firstName;
+
+    @NotBlank(message = "Last name can't be blank")
+    @Column(name = "last_name")
+    private String lastName;
+
+    @Email(message = "Email must be valid")
+    private String email;
+
+    @NotBlank
+    private String password;
+
+    @ManyToOne
+    private Role role;
```

Comment:
I would explicitly define the join column:
`    @JoinColumn(name = "role_id", referencedColumnName = "id", updatable = false)
`